common/test/TestLogging.cs

```diff
@@ -2,39 +2,114 @@
+                        };
+                        _telemetryClient = new TelemetryClient(config);
+                        _testRunId = Guid.NewGuid().ToString();
+                    }
+                }
+            }
         }
 
         private const string NullInstance = "(null)";
-        
+
         public static string IdOf(object value) => value != null ? value.GetType().Name + "#" + GetHashCode(value) : NullInstance;
 
         public static int GetHashCode(object value) => value?.GetHashCode() ?? 0;
 
-        public static TestLogging GetInstance()
+        public static TestLogging GetInstance(TestContext testContext = null)
         {
-            return s_instance;
+            var logger = new TestLogging();
+
+            if (_telemetryClient != null)
+            {
+                // Set logger properties for the current test.
+                var targetFramework = (TargetFrameworkAttribute)Assembly.GetExecutingAssembly()
+                .GetCustomAttributes(typeof(TargetFrameworkAttribute), false)
+                .SingleOrDefault();
+                logger.Context = testContext;
+                logger.TargetFramework = targetFramework.FrameworkName;
+            };
+
+            return logger;
         }
 
-        public void WriteLine(string message)
+        public void WriteLine(string message, SeverityLevel severity = SeverityLevel.Information, IDictionary<string, string> propertyBag = null, [CallerMemberName] string caller = null)
         {
+            // Log to event source
             EventSourceTestLogging.Log.TestMessage(message);
+
+            // Log to Application insights
+            if (_telemetryClient != null)
+            {
+                if (propertyBag == null)
+                {
+                    propertyBag = new Dictionary<string, string>();
+                }
+
+                // Add common properties
+                propertyBag.Add(TelemetryPropertyNames.TestRunId, _testRunId);
+                propertyBag.Add(TelemetryPropertyNames.TestName, Context?.TestName);
+                propertyBag.Add(TelemetryPropertyNames.ClassName, Context?.FullyQualifiedTestClassName ?? caller);
+                propertyBag.Add(TelemetryPropertyNames.TestFramework, TargetFramework);
+                propertyBag.Add(TelemetryPropertyNames.TestStatus, Context?.CurrentTestOutcome.ToString());
+                propertyBag.Add(TelemetryPropertyNames.Language, TelemetryConstants.Language);
+                propertyBag.Add(TelemetryPropertyNames.Service, TelemetryConstants.Service);
+
+                // Environment Variable set in Azure pipelines to correlate logs with the build number.
+                propertyBag.Add("BuildId", Environment.GetEnvironmentVariable("BUILD_BUILDID"));
+
+                _telemetryClient.TrackTrace(message, severity, propertyBag);
+            }
         }
 
-        public void WriteLine(string format, params object[] args)
```

Comment:
was this method call unused?

Reply:
it was unused

Reply:
I want to rename WriteLine to something else in the next PR and if we need one with args, we can add an override with correct name when needed.